flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/logical/LegacyTypeInformationType.java

```diff
@@ -65,12 +65,20 @@ public LogicalType copy(boolean isNullable) {
 
 	@Override
 	public String asSerializableString() {
-		throw new TableException("Legacy type information has no serializable string representation.");
+		return withNullability(
+			FORMAT,
+			getTypeRoot(),
+			typeInfo.getTypeClass().getName(),
+			EncodingUtils.encodeObjectToString(typeInfo));
```

Comment:
How about still use `TypeStringUtils`?

Reply:
Do you mean use `TypeStringUtils` to replace `EncodingUtils.encodeObjectToString(typeInfo)`? In this way, we have to escape the string.

Reply:
Yes, I think it is good to read.